File: test_storage.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv(r'f:\prodash-nextweb\.env.local')

url: str = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key: str = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

if not key:
    key = os.environ.get("NEXT_PUBLIC_SUPABASE_ANON_KEY")

supabase: Client = create_client(url, key)

res = supabase.storage.from_('firmwares').list()
print(res)

File: test_storage.js
const { createClient } = require("@supabase/supabase-js");
require("dotenv").config({ path: ".env.local" });

const url = process.env.NEXT_PUBLIC_SUPABASE_URL;
const key = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

const supabase = createClient(url, key);

async function test() {
    const { data, error } = await supabase.storage.from("firmwares").list();
    console.log("Data:", data);
    console.log("Error:", error);
}

test();
